C/casacore/build_tarballs.jl

```diff
@@ -0,0 +1,89 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "casacore"
+version = v"3.4.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/casacore/casacore.git", "a735bf5f31ea012ac4f7f7378a1f89c1fc136a06")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+OPENBLAS=(-lopenblas64_)
+
+cd $WORKSPACE/srcdir/casacore
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBUILD_PYTHON=OFF \
+      -DBUILD_SHARED_LIBS=ON \
+      -DUSE_READLINE=OFF \
+      -DUSE_HDF5=ON \
+      -DFFTW3_DISABLE_THREADS=ON \
+      -DBLAS_LIBRARIES="${OPENBLAS[*]}" \
+      ..
+make -j${nproc}
+make install
+exit
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms(; experimental=true))
```

Comment:
Ugh, this depends on HDF5, so we're constrained by the available HDF5 binaries.  Also:
```
┌ Warning: /agent/_work/1/s/C/casacore/build/x86_64-linux-gnu-cxx11/Jip4jYeI/x86_64-linux-gnu-libgfortran3-cxx11/destdir/lib/libcasa_scimath_f.so.6 links to libgfortran!  This causes incompatibilities across major versions of GCC.  To remedy this, you must build a tarball for each major version of GCC.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_gfortran_versions(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/EHPCr/src/auditor/compiler_abi.jl:60
```
```suggestion
platforms = [
    Platform("x86_64", "linux"),
    Platform("aarch64", "linux"; libc="glibc"),
    Platform("x86_64", "macos"),
    Platform("x86_64", "windows"),
    Platform("i686", "windows"),
]
platforms = expand_gfortran_versions(expand_cxxstring_abis(platforms))
```

Reply:
Yeah - I think I can disable the feature? It seems that is the biggest bottleneck in targets. Also, Bison doesn't allow for windows builds it seems.

Reply:
You mean disable HDF5?  How much important is it?

Reply:
I mean, it provides hdf5 serialization for all the main types in casa. I suppose in the wrapper I inevitably write, I could just call out to HDF5.jl and package the important bits.

Reply:
Makes sense :slightly_smiling_face: